manager/map.go

```diff
@@ -178,7 +177,21 @@ func (m *Map) close(cleanup MapCleanupType) error {
 		if m.PinPath != "" {
 			err = ConcatErrors(err, os.Remove(m.PinPath))
 		}
-		return ConcatErrors(err, m.array.Close())
+		err = ConcatErrors(err, m.array.Close())
+		if err != nil {
+			return err
+		}
+		m.cleanup()
```

Comment:
Shouldn't we do :
```
if shouldClose {
    defer m.cleanup()
    ...
```

Reply:
Mmm ... I'd rather not reset the map if `m.array.Close()` fails: if possible it would be best to keep the references to the actual underlying `ebpf.Map` and `ebpf.MapSpec` to know which map it is.